torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -113,7 +118,7 @@ struct TORCH_API AccumulateGrad : public Node {
       if (!GradMode::is_enabled() && !new_grad.is_sparse() &&
           !new_grad.is_sparse_csr() &&
           !(variable.is_sparse_csr() && new_grad.layout() == at::kStrided) &&
-          new_grad.use_count() <= num_expected_refs &&
+          torch::autograd::adjusted_use_count(new_grad) <= num_expected_refs &&
```

Comment:
Why is this ok?
Once this get returned to the user, they are free to do whatever they want with it. In particular any inplace update! So you would need to properly invalidate any caching related to this Tensor (which you can't do).

Reply:
If a user does caching of a TensorImpl outside of our system, that is a general limitation of cudagraphs, and not related to this pr

Reply:
But this check is to choose if the TensorImpl is used as-is in the `.grad` field. So it is expected that the user will use this .grad outside of your system!

Reply:
For both warmup and recording, we are hitting this path, and for both of those we have the same limitation that we outputs from one generation cannot be present for the next generation. So this is already the current state. We could try ban accumulation for cudagraphs outputs but it's not clear that the new grad wouldn't be a cudagraph output either. I would prefer to do that in a separate PR.

Reply:
Would it be cleaner if you instead adjusted num_expected_refs at the callsite?